Android Youtube Refactor

devon-o opened the pull request:
Made better use of States for checking disposal.
Created views in initVideo rather than using lazy instantiation throughout.
Used interface for custom youtube fragment callbacks rather than calling methods on static extension.context.
Cleanup of dispose().

petermd committed:
refactor

bmarsh commented on android/src/meez/nativeExtensions/androidyoutube/AndroidYoutubeContext.java:
```diff
@@ -168,21 +167,55 @@ public void dispose()
+        });
 
-        //Hack - videoContainer is only added to view long enough to attach fragment.
+        //Hack - videoContainer is only added to root view long enough to attach fragment (fragment must be attached to view in Android 'display list').
```
Is there no alternative to moving the fragment over to a dialog? Can we not absolute position a fragment within `videoContainer`?

devon-o replied:
The dialog was added as I didn't know a way of absolutely positioning a FrameLayout (or a fragment within a FrameLayout). I am open to any ideas to get rid of that.

devon-o committed:
Cleaned up State Machine

Updated State Machine.
Stopped setting everything to null on dispose.
Logging warning in checkState.
Added IDE Files and Dirs to gitignore

bmarsh commented:
LGTM
